### ISSUE TITLE
Dataloader get 5x slow down at the 2nd epoch on CPU with augment=True

### ISSUE THREAD

**Issue by xhzhao** [SeanNaren/deepspeech.pytorch#310]
hi i'm trying to run this workload on CPU, and i found that the dataloader takes too much time in the 2nd epoch. The dataset is AN4, and the augment=True, num_worker=1. The hotspot in dataloader is the sox command: https://github.com/SeanNaren/deepspeech.pytorch/blob/master/data/data_loader.py#L297 
I get the average time on this line:
- 1st epoch average: 0.0261 s
- 2nd epoch average: 0.1255 s
I double checked that GPU have no such issue.
I understand that one new dataloader thread will be created in the 2nd epoch, and the augmented file will be saved at the /tmp/ folder. But i really don't know why this sox command takes so much time in the second epoch!!!

**Comment by ryanleary** [comment]
I'm not sure why it would be different in the second epoch vs the first. Perhaps there's some I/O contention? You should probably increase the number of workers as well.

**Comment by miguelvr** [comment]
There are 2 separate problems here:

 1) 2nd epoch is slower - actually it's not only the 2nd epoch that is slower, it's the 2nd epoch onwards that is slower than the 1st epoch. This is due to the sortagrad technique used: in the first epoch the samples are batched by similar sample lengths, leading to minimal padding. This means that on average, each batch has less "data". 

2) Usage of `--augment` - yes, this slows down training substantially because it's performing random variations to the audio when it loads it every time. You notice it much more on the GPU because the `sox` augmentation operations are done on the CPU and then have to be loaded into the GPU, which drastically slows down training. The solution for this would be to do augmentation once, before training, and then create a manifest with the augmented samples.

Hope this helps,
Miguel

**Comment by SeanNaren** [comment]
Thanks @miguelvr I think that covers it :)
